dashboard/app/models/pd/fit_weekend_registration_base.rb

```diff
@@ -16,5 +16,139 @@
 #
 
 class Pd::FitWeekendRegistrationBase < ActiveRecord::Base
+  include Pd::Form
+
   self.table_name = 'pd_fit_weekend_registrations'
+
+  belongs_to :pd_application, class_name: 'Pd::Application::ApplicationBase'
+
+  after_initialize :set_registration_year
+  before_validation :set_registration_year
+
+  after_create :update_application_status
+  def update_application_status
+    pd_application.update!(status: 'withdrawn') unless accepted?
+  end
+
+  after_create :send_fit_weekend_confirmation_email
+  def send_fit_weekend_confirmation_email
+    Pd::FitWeekend1819RegistrationMailer.confirmation(self).deliver_now
+  end
+
+  # Override in derived classes and set to valid value.
+  # Setting to nil here fails those validations and prevents this base class from being saved.
+  def set_registration_year
+    self.registration_year = nil
+  end
+
+  YES = 'Yes'.freeze
+  NO = 'No'.freeze
+  YES_OR_NO = [YES, NO].freeze
+
+  def self.options
+    {
+      ableToAttend: YES_OR_NO,
+      dietaryNeeds: [
+        'None',
+        'Vegetarian',
+        'Vegan',
+        'Kosher',
+        'Halal',
+        'Gluten Free',
+        'Food Allergy',
+      ],
+      liveFarAway: YES_OR_NO,
+      addressState: get_all_states_with_dc.to_h.values,
+      howTraveling: [
+        'I will drive by myself',
+        'I will carpool with another FiT Weekend attendee (Please note who)',
+        'Flying',
+        'Amtrak or regional train service',
+        'Public transit (e.g., city bus or light rail)',
+      ],
+      needHotel: YES_OR_NO,
+      needAda: YES_OR_NO,
+      photoRelease: [YES],
+      liabilityWaiver: [YES],
+    }.freeze
+  end
+
+  def self.required_fields
+    [
+      :preferred_first_name,
+      :last_name,
+      :email,
+      :phone,
+      :able_to_attend,
+      :contact_first_name,
+      :contact_last_name,
+      :contact_relationship,
+      :contact_phone,
+      :dietary_needs,
+      :live_far_away,
+      :how_traveling,
+      :need_hotel,
+      :photo_release,
+      :liability_waiver,
+      :agree_share_contact,
+    ].freeze
+  end
+
+  def validate_required_fields
+    hash = sanitize_form_data_hash
+
+    if hash.try(:[], :able_to_attend) == NO
+      # then we don't care about the rest of the fields
+      return
+    end
+
+    super
```

Comment:
There isn't really a super class to invoke here. This should be removed. If there are fields that we'll always want, they should be here. Otherwise this should be a method that expects a derived class to override and throws an exception if it doesn't

Reply:
now this just calls the super class (Pd::Form)'s method. Derived classes can add additional validations.